### PR TITLE
 refactor cli/dashboard/apiserver to reference folder

### LINKED ISSUE #1030: [code refactor] move cli/apiserver/dashboard related code from `pkg/` into `referrence/` folder
**Is your feature request related to a problem? Please describe.**

We're going to [release 1.0](https://github.com/oam-dev/kubevela/issues/932) soon. KubeVela 1.0 will mainly focus on platform builder, and core features are all platform builder centric, you can refer to  https://github.com/oam-dev/kubevela/issues/922 for high level concept. At that time, these core features will be in GA status.

While the current implementation of vela cli, apiserver and dashboard will not be good enough to be called GA. And that should not affect our core features. 

On the other hand, we're implementing `velacp`(https://github.com/oam-dev/velacp) as a separated control plane which will become the second generation of apiserver of kubevela. #987 

**Describe the solution you'd like**

So I propose we should refactor our code sturcture to move cli/apiserver/dashboard related code from `pkg/` into `referrence/` folder.

Separate the CI tests out, run both of these tests but only calculate the test coverage for the core part.

We can also raise our test coverage bar for the core part, I suggest at least 70% unit test coverage is required.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by wonderflow**
fix #1030

### CODE REVIEW

File: references/appfile/addon.go [hunk 1]
```diff
@@ -9,16 +9,18 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/oam-dev/kubevela/pkg/appfile"
```
Comment (ryanzhang-oss) [inline references/appfile/addon.go:12]: please regroup

File: references/cli/logs.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 	"text/template"
 	"time"
 
+	appfile2 "github.com/oam-dev/kubevela/references/appfile"
```
Comment (ryanzhang-oss) [inline references/cli/logs.go:11]: why appfile2? 
please regroup the imports

File: references/cli/ls.go [hunk 3]
```diff
@@ -1,18 +1,19 @@
-package commands
+package cli
 
 import (
 	"context"
 	"strings"
 
+	appfile2 "github.com/oam-dev/kubevela/references/appfile"
```
Comment (ryanzhang-oss) [inline references/cli/ls.go:7]: why appfile2?

please regropu the goimports

File: references/common/trait.go [hunk 4]
```diff
@@ -1,22 +1,22 @@
-package serverlib
+package common
 
 import (
 	"context"
 	"fmt"
 	"strings"
 
-	"github.com/oam-dev/kubevela/pkg/appfile/api"
+	appfile2 "github.com/oam-dev/kubevela/references/appfile"
```
Comment (ryanzhang-oss) [inline references/common/trait.go:8]: same here

File: references/common/workload.go [hunk 5]
```diff
@@ -1,20 +1,21 @@
-package serverlib
+package common
 
 import (
 	"context"
 	"fmt"
 	"strconv"
 	"strings"
 
+	appfile2 "github.com/oam-dev/kubevela/references/appfile"
```
Comment (ryanzhang-oss) [inline references/common/workload.go:9]: same

File: references/plugins/cluster.go [hunk 6]
```diff
@@ -15,13 +15,13 @@ import (
...
 	corev1alpha2 "github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
 	"github.com/oam-dev/kubevela/apis/types"
-	util2 "github.com/oam-dev/kubevela/pkg/commands/util"
 	"github.com/oam-dev/kubevela/pkg/cue"
 	"github.com/oam-dev/kubevela/pkg/oam/discoverymapper"
 	"github.com/oam-dev/kubevela/pkg/oam/util"
 	"github.com/oam-dev/kubevela/pkg/utils/common"
 	"github.com/oam-dev/kubevela/pkg/utils/helm"
 	"github.com/oam-dev/kubevela/pkg/utils/system"
+	util2 "github.com/oam-dev/kubevela/pkg/utils/util"
```
Comment (ryanzhang-oss) [inline references/plugins/cluster.go:24]: no one uses this?

Comment (wonderflow) [reply references/plugins/cluster.go:24]: used

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**